### PR TITLE
 SONARJAVA-4540 Support Record's Compact Constructors in MethodTreeImpl

### PR DESCRIPTION
**Pull request by erwan-serandour-sonarsource**

### CODE REVIEW

File: java-frontend/src/test/java/org/sonar/java/model/declaration/ClassTreeImplTest.java [hunk 1]
```diff
@@ -87,6 +87,7 @@ public void visitVariable(VariableTree tree) {
   void records_members_order() {
     ClassTree classTree = (ClassTree) JParserTestUtils.parse(
       "record Output(String title, String summary, String text) {\n"
+        + "  public Output {}"
```
Comment (dorian-burihabwa-sonarsource) [inline java-frontend/src/test/java/org/sonar/java/model/declaration/ClassTreeImplTest.java:90]: Nice addition, it is a shame we did not test compact constructors before

File: java-frontend/src/test/java/org/sonar/java/model/declaration/ClassTreeImplTest.java [hunk 2]
```diff
@@ -100,12 +101,16 @@ void records_members_order() {
...
       "VARIABLE",
       "METHOD",
       "VARIABLE",
       "CONSTRUCTOR",
       "VARIABLE",
       "CLASS",
       "VARIABLE");
+
+    MethodTreeImpl constructor = (MethodTreeImpl) classTree.members().get(0);
+    assertThat(constructor.getLine()).isEqualTo(2);
```
Comment (dorian-burihabwa-sonarsource) [inline java-frontend/src/test/java/org/sonar/java/model/declaration/ClassTreeImplTest.java:114]: I think this assertion should be tested in a dedicated test

File: java-frontend/src/main/java/org/sonar/java/model/declaration/MethodTreeImpl.java [hunk 3]
```diff
@@ -233,7 +234,16 @@ public void accept(TreeVisitor visitor) {
...
+    InternalSyntaxToken token = parameters.openParenToken();
+    if(token != null){
+      return token.getLine();
+    }else{
+      //type cast may fail, it is fine. We will just add a new case if it happens.
+      //could first try with type cast and fallback parameters
+      //but cannot reach full coverage
+      InternalSyntaxToken name = (InternalSyntaxToken)simpleName().identifierToken();
+      return name.getLine();
+    }
```
Comment (dorian-burihabwa-sonarsource) [inline java-frontend/src/main/java/org/sonar/java/model/declaration/MethodTreeImpl.java:246]: We should format the code to make it more readable

File: java-frontend/src/test/java/org/sonar/java/model/declaration/ClassTreeImplTest.java (hunk 2 above)
Comment (dorian-burihabwa-sonarsource) [reply java-frontend/src/test/java/org/sonar/java/model/declaration/ClassTreeImplTest.java:114]: We could probably move this to `MethodTreeImplTest.java`

File: java-checks/src/test/files/checks/IndentationCheck_default.java [hunk 4]
```diff
@@ -1,6 +1,18 @@
...
+      public QonId { // Noncompliant
+      Objects.requireNonNull(value, "QonId value cannot be null");
+    }
+  }
+
+  public record QonId_(String value) {
+    public QonId_ { // compliant
+      Objects.requireNonNull(value, "QonId value cannot be null");
+    }
+  }
```
Comment (dorian-burihabwa-sonarsource) [inline java-checks/src/test/files/checks/IndentationCheck_default.java:15]: It looks like we added this test case to a non-compiling file. Let's move that file to `java-checks-test-sources/src/main/java/checks` or at least `java-checks-test-sources/src/main/files/non-compiling/checks`

**Review by dorian-burihabwa-sonarsource — CHANGES_REQUESTED** [review]
The change looks good, let's make sure we place our tests in the right location!

File: java-frontend/src/main/java/org/sonar/java/model/declaration/MethodTreeImpl.java (hunk 3 above)
Comment (erwan-serandour-sonarsource) [reply java-frontend/src/main/java/org/sonar/java/model/declaration/MethodTreeImpl.java:246]: :+1:

File: java-checks/src/test/files/checks/IndentationCheck_default.java (hunk 4 above)
Comment (erwan-serandour-sonarsource) [reply java-checks/src/test/files/checks/IndentationCheck_default.java:15]: :+1:

File: java-frontend/src/test/java/org/sonar/java/model/declaration/ClassTreeImplTest.java (hunk 1 above)
Comment (erwan-serandour-sonarsource) [reply java-frontend/src/test/java/org/sonar/java/model/declaration/ClassTreeImplTest.java:90]: :+1:

File: java-frontend/src/test/java/org/sonar/java/model/declaration/ClassTreeImplTest.java (hunk 2 above)
Comment (erwan-serandour-sonarsource) [reply java-frontend/src/test/java/org/sonar/java/model/declaration/ClassTreeImplTest.java:114]: Seemed also weird to me, but did not know where to put it.

Comment (erwan-serandour-sonarsource) [reply java-frontend/src/test/java/org/sonar/java/model/declaration/ClassTreeImplTest.java:114]: :+1:

**Review by dorian-burihabwa-sonarsource — CHANGES_REQUESTED** [review]
Thanks for the changes! To finish the work, let's move **all** the test sources to `java-checks-test-sources`

**Review by dorian-burihabwa-sonarsource — APPROVED** [review]
Thanks for the changes!

**Pull request merged**